slack-web.cabal

```diff
@@ -25,8 +25,73 @@ extra-source-files:
   CHANGELOG.md
   README.md
 
+common build-opts
+  ghc-options:
+    -Weverything
+    -- missing-exported-signatures turns off the more strict -Wmissing-signatures. See https://ghc.haskell.org/trac/ghc/ticket/14794#ticket
+    -Wno-missing-exported-signatures
+    -- Requires explicit export lists for every module, a pain for large modules
+    -Wno-missing-export-lists
+    -- Requires explicit imports of _every_ function (e.g. '$'); too strict
+    -Wno-missing-import-lists
```

Comment:
In Purescript they thought about this. They allow you to have exactly zero or one unqualified import that doesn't have an explicit import list 😆

Reply:
clever!

I just copied the Mercury list :p